### ISSUE TITLE
Incorrect calculation of steam enthalpy in 2-phase region with v5.1.2

### ISSUE THREAD

**Issue by matthew-yates** [CoolProp/CoolProp#943]
If you run the following Python code:
```
import CoolProp as CP
T1 = 323
S1 = 5706
H1 = CP.CoolProp.PropsSI('H','T',T1,'S',S1,'Water')
Qual = CP.CoolProp.PropsSI('Q','T',T1,'S',S1,'Water')
HsatV = CP.CoolProp.PropsSI('H','T',T1,'Q',1,'Water')
HsatL = CP.CoolProp.PropsSI('H','T',T1,'Q',0,'Water')
H2 = HsatV*Qual + HsatL*(1-Qual)
print("The quality is:",round(Qual*100,1),"%")
print("The enthalpy calculated from quality and saturated enthalpies:",round(H2,2),"J/kg")
print("The enthalpy calculated by CoolProp at the same state is:",round(H1,2),"J/kg")
```
Then the output is:
```
The quality is: 67.8 %
The enthalpy calculated from quality and saturated enthalpies is: 1825049.34 J/kg
The enthalpy calculated by CoolProp at the same state is: 2589240.92 J/kg
```
 H1 does not equal H2, although it should.  I'm a new user, so I apologize if I am missing something.

**Comment by ibell** [comment]
Works fine for me:
```
from __future__ import print_function
import CoolProp as CP
print("version:",CP.__version__)
print("gitrevision:",CP.__gitrevision__)
T1 = 323
S1 = 5706
H1 = CP.CoolProp.PropsSI('H','T',T1,'S',S1,'Water')
Qual = CP.CoolProp.PropsSI('Q','T',T1,'S',S1,'Water')
HsatV = CP.CoolProp.PropsSI('H','T',T1,'Q',1,'Water')
HsatL = CP.CoolProp.PropsSI('H','T',T1,'Q',0,'Water')
H2 = HsatV*Qual + HsatL*(1-Qual)
print("The quality is:",round(Qual*100,1),"%")
print("The enthalpy calculated from quality and saturated enthalpies:",round(H2,2),"J/kg")
print("The enthalpy calculated by CoolProp at the same state is:",round(H1,2),"J/kg")
```
yields
```
version: 5.1.3dev
gitrevision: 8fdf0b92c1ec0f7fa6f43f50a9728cf75600c098
The quality is: 67.8 %
The enthalpy calculated from quality and saturated enthalpies: 1825049.34 J/kg
The enthalpy calculated by CoolProp at the same state is: 1825049.34 J/kg
```
What version of CoolProp are you using?  What operating system?

**Comment by matthew-yates** [comment]
Thanks for the reply.  I'm not sure why I'm not getting the same output.  I ran the following:
```
from __future__ import print_function
import CoolProp as CP
import sys
print(sys.version)
print("version:",CP.__version__)
print("gitrevision:",CP.__gitrevision__)
T1 = 323
S1 = 5706
H1 = CP.CoolProp.PropsSI('H','T',T1,'S',S1,'Water')
Qual = CP.CoolProp.PropsSI('Q','T',T1,'S',S1,'Water')
HsatV = CP.CoolProp.PropsSI('H','T',T1,'Q',1,'Water')
HsatL = CP.CoolProp.PropsSI('H','T',T1,'Q',0,'Water')
H2 = HsatV*Qual + HsatL*(1-Qual)
print("The quality is:",round(Qual*100,1),"%")
print("The enthalpy calculated from quality and saturated enthalpies:",round(H2,2),"J/kg")
print("The enthalpy calculated by CoolProp at the same state is:",round(H1,2),"J/kg")
```
Here is the output:
```
3.5.1 |Anaconda 2.4.1 (64-bit)| (default, Dec  7 2015, 11:16:01) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]
version: 5.1.2
gitrevision: d9132da24b171e11a9cf1a59322d6b10075238d6
The quality is: 67.8 %
The enthalpy calculated from quality and saturated enthalpies: 1825049.34 J/kg
The enthalpy calculated by CoolProp at the same state is: 2589240.92 J/kg
```
The code was run on my laptop running 64-bit Linux (Mint version 17.3).  I've installed Python 3.5.1 through Anaconda.  CoolProp was installed using pip.  The code was run in a Jupyter notebook.   I'm not experienced with Python, so hopefully I haven't done anything dumb.

**Comment by ibell** [comment]
That's very weird.  I wonder if for some reason it's a python 3.5+cython
issue.  I don't use python 3.5.  Let me have a go with that.

**Comment by ibell** [comment]
Could you perhaps try with python 2.7?  I'm having a hard time getting
python 3.5 running on my window machine.

**Comment by matthew-yates** [comment]
I did some more research, and the problem has something to do with the CoolProp version 5.1.2 installed automatically using pip.  Below is a summary of stuff I tried (in the order I tried it, although maybe it doesn't matter).

First I created a Python 2.7 environment in Anaconda, ran the above code again, and I get the same result.  Here is the output of the code:
```
2.7.11 |Anaconda 2.4.1 (64-bit)| (default, Dec  6 2015, 18:08:32) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]
version: 5.1.2
gitrevision: d9132da24b171e11a9cf1a59322d6b10075238d6
The quality is: 67.8 %
The enthalpy calculated from quality and saturated enthalpies: 1825049.34 J/kg
The enthalpy calculated by CoolProp at the same state is: 2589240.92 J/kg
```
Next I created a Python 2.6 environment, ran the code again, and got the same result.  Here is the output:
```
2.6.9 |Anaconda 2.3.0 (64-bit)| (unknown, Aug 21 2014, 18:28:52) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]
version: 5.1.2
gitrevision: d9132da24b171e11a9cf1a59322d6b10075238d6
The quality is: 67.8 %
The enthalpy calculated from quality and saturated enthalpies: 1825049.34 J/kg
The enthalpy calculated by CoolProp at the same state is: 2589240.92 J/kg
```
I noticed each time I reinstalled CoolProp for the new environment, it was using the cached version.  I reactivated the 2.7 environment, uninstalled CoolProp, then manually installed the latest version of CoolProp using:
```
git clone https://github.com/CoolProp/CoolProp --recursive
cd CoolProp/wrappers/Python
python setup.py install --user
```
I ran the code again.  This time I got the correct output:
```
2.7.11 |Anaconda 2.4.1 (64-bit)| (default, Dec  6 2015, 18:08:32) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]
version: 5.1.3dev
gitrevision: ed8f1a9d1e4cd434e802fb305653368b5decad2e
The quality is: 67.8 %
The enthalpy calculated from quality and saturated enthalpies: 1825049.34 J/kg
The enthalpy calculated by CoolProp at the same state is: 1825049.34 J/kg
```
I went back to my Python 3.5 environment, uninstalled CoolProp 5.1.2 using pip, manually installed the newer version with:
```
cd CoolProp/wrappers/Python
python setup.py install --user
```
I ran the code, and now get the correct output in Python 3.5.1.  Here is the output:
```
3.5.1 |Anaconda 2.4.1 (64-bit)| (default, Dec  7 2015, 11:16:01) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]
version: 5.1.3dev
gitrevision: ed8f1a9d1e4cd434e802fb305653368b5decad2e
The quality is: 67.8 %
The enthalpy calculated from quality and saturated enthalpies: 1825049.34 J/kg
The enthalpy calculated by CoolProp at the same state is: 1825049.34 J/kg
```

**Comment by ibell** [comment]
Weird.  I've found sometimes that I get more reliable results if I do a few ``pip uninstall coolprop`` until it claims that it can't uninstall it anymore.  And then I install.

**Comment by ibell** [comment]
Anyway, glad to hear its working, I have no idea why you saw that behavior before.

**Comment by ibell** [comment]
Interestingly on OSX with python 3.5, with a pip install of CoolProp, I get
```
(py35)Ians-Mac-mini:Desktop ian$ python check_Ts.py 
version: 5.1.2
gitrevision: d9132da24b171e11a9cf1a59322d6b10075238d6
The quality is: 67.8 %
The enthalpy calculated from quality and saturated enthalpies: 1825049.34 J/kg
The enthalpy calculated by CoolProp at the same state is: 2589240.92 J/kg
```

**Comment by matthew-yates** [comment]
Hmm.  Exactly the same wrong number for enthalpy that I got.  I'll make sure to test it on my desktop at work tomorrow and let you know if I discover any strange behaviour.

Thanks for making this tool available.  I'm developing some course material, and CoolProp is very convenient for teaching about heat engines and power cycles.

**Comment by ibell** [comment]
So something is clearly messed up with the ``pip`` version of CoolProp released with 5.1.2.

**Comment by ibell** [comment]
I think we aren't far from 5.1.3, maybe that will resolve whatever is causing this.

**Comment by ibell** [comment]
Same on windows with python 2.7
```
[py27cp512] C:\Users\Belli\Desktop>python sample.py
version: 5.1.2
gitrevision: d9132da24b171e11a9cf1a59322d6b10075238d6
The quality is: 67.8 %
The enthalpy calculated from quality and saturated enthalpies: 1825049.34 J/kg
The enthalpy calculated by CoolProp at the same state is: 2589240.92 J/kg
```
So weird.  Looks like 5.1.3 will fix whatever is causing this behavior.

**Comment by ibell** [comment]
@matthew-yates You should send an email to the mailing list for coolprop.  There are a number of other professors also using CoolProp for teaching purposes.

**Comment by ibell** [comment]
Will be fixed in version 5.1.3
